### PR TITLE
use ConnectivityMeasure for GraphLasso

### LINKED ISSUE #1213: sklearn's GraphLassoCV or ConnectivityMeasure(cov_estimator=GraphLassoCV) ?
Many examples fit  `sklearn.covariance.GraphLassoCV`. We can rely on `ConnectivityMeasure` instead. One drawback is that we will not be able to assess the `GraphLasso` attributes, such as the `alpha_`.

### PR DESCRIPTION
**Pull request by salma1601**
fixes #1213.
For the moment I am keeping direct fitting of sklearn GraphLasso in `plot_simulated_connectome` because of `alpha_` parameter use, and will do another PR for this example.

### CODE REVIEW

**Comment by AlexandreAbraham** [comment]
LGTM

File: examples/03_connectivity/plot_inverse_covariance_connectome.py [hunk 1]
```diff
@@ -49,9 +49,14 @@
...
-estimator = GraphLassoCV()
+from nilearn.connectome import ConnectivityMeasure
+covariance_measure = ConnectivityMeasure(cov_estimator=GraphLassoCV(),
+                                         kind='covariance')
+covariance_matrix = covariance_measure.fit_transform([time_series])[0]
 
-estimator.fit(time_series)
+precision_measure = ConnectivityMeasure(cov_estimator=GraphLassoCV(),
+                                        kind='precision')
+precision_matrix = covariance_measure.fit_transform([time_series])[0]
```
Comment (GaelVaroquaux) [inline examples/03_connectivity/plot_inverse_covariance_connectome.py]: That example just got much more complex. I am not too excited about that.

File: examples/03_connectivity/plot_multi_subject_connectome.py [hunk 2]
```diff
@@ -89,26 +89,33 @@ def plot_matrices(cov, prec, title):
...
+from sklearn.covariance import GraphLassoCV
+gl_covariance_measure = ConnectivityMeasure(GraphLassoCV(verbose=2),
+                                            kind='covariance')
+gl_covariance = gl_covariance_measure.fit_transform(
+    [np.concatenate(subject_time_series)])[0]
+
+gl_precision_measure = ConnectivityMeasure(GraphLassoCV(verbose=2),
+                                           kind='precision')
+gl_precision = gl_covariance_measure.fit_transform(
+    [np.concatenate(subject_time_series)])[0]
```
Comment (GaelVaroquaux) [inline examples/03_connectivity/plot_multi_subject_connectome.py]: This is also adding a lot of lines. I am a bit worried that it actually makes all this harder to explain / learn.

**Comment by salma1601** [comment]
What I don't like is that the `ConnectivityMeasure` object has already a covariance estimator in it, and then in examples we never benefit from this estimator and do things manually.
I don't mean to insist I am rather a humble user but I don't like when 2 objects do the same thing: it puzzles me and I become hesitating on what to use when. This for me is more important than the number of lines.

Also, I want to do another PR removing all the precision/covariances and using their normalized versions, and this is easier to do with `ConnectivityMeasure`.
@GaelVaroquaux  May be I can do one unique PR to solve this + #1214 + #1223 so that you have a more global picture and then decide to merge or not ?

**Comment by GaelVaroquaux** [comment]
> What I don't like is that the ConnectivityMeasure object has already a
> covariance estimator in it, and then in examples we never benefit from this
> estimator and do things manually.
> I don't mean to insist I am rather a humble user but I don't like when 2
> objects do the same thing: it puzzles me and I become hesitating on what to use
> when. This for me is more important than the number of lines.

We should add a note in the examples that use the raw covariance
estimators. However, I've been teaching yesterday, and it seems to me
that the simplicity is the most important thing.

> Also, I want to do another PR removing all the precision/covariances and using
> their normalized versions, and this is easier to do with ConnectivityMeasure.
> @GaelVaroquaux May be I can do one unique PR to solve this + #1214 + #1223 so
> that you have a more global picture and then decide to merge or not ?

Normalizing is indeed a benefit. I am however still a bit worried by the
complexity. It's not only the number of the lines, it's that we juggle
between lists and single-subject elements, it's that we import from
multiple packages (imports confuse beginners)...

**Comment by salma1601** [comment]
Sorry, but I am going to insist, because I am still not convinced :)

> we juggle between lists and single-subject elements

A bit similarly as when `np.concatenate(subject_time_series)` is used. Note that with `ConnectivityMeasure` we can do this without concatenating the subjects.

> we import from multiple packages

We can add a bit of narration when we import from sklearn, saying that we rely on its covariance estimators.

>I've been teaching yesterday, and it seems to me that the simplicity is the most important thing.

Yes I agree, and the lazy nilearn user that I am does not know that much about sklearn API :$.
It can be a goal to push people to go to sklearn covariance estimators and understand their methods and attributes. Personally I prefer they understand how the `ConnectivityMeasure` works, that it fits multiple subjects at once, and that it hides a covariance estimator object that we can just not care about.

>Normalizing is indeed a benefit

Yes, and many people have never ever heard about precision matrix, but know partial correlations, what they model, and how to do statistical tests on them. I would love to just throw away that precision matrix!

**Comment by kchawla-pi** [comment]
Is this alive? If not, please close it by this weekend.
